app/controllers/orders_controller.rb

```diff
@@ -8,20 +8,38 @@ def new
 
   def create
     @order = current_user.orders.build(order_params)
+    cart_items = @cart.cart_items.includes(:product)
 
+    cart_items.each do |cart_item|
+      order_detail = @order.order_details.build(
+        product: cart_item.product
+      )
+    end
     if @order.save
-      OrderMailer.order_confirmation(@order).deliver_now
+      Rails.cache.delete('most_ordered_product')
       current_user.cart.cart_items.destroy_all
       redirect_to products_path, notice: 'Order successfully placed.'
     else
       redirect_to products_path, alert: 'Order creation failed.'
     end
   end
 
+  def show
+    @order = Order.find(params[:id])
+    @product_details = @order.product_details
+  end
+
   def admin_index
     @orders = Order.all
   end
 
+  def most_ordered_product
+    most_ordered_product = Rails.cache.fetch('most_ordered_product', expires_in:1.hour) do
+      Product.joins(:order_details).group('products.id').order('COUNT(order_details.id) DESC').first
+    end
+    @most_ordered_product = most_ordered_product
```

Comment:
no need for extra variable most_ordered_product. You can simply assign cached result to @most_ordered_product